### PR TITLE
add tim reveiwer

### PR DESCRIPTION
**Pull request by edik24**
adding the option to automatically assign tim reviewer to PR's where tim items (feed or indicators) are affected.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by edik24** [comment]
example of open PR is at:
https://github.com/demisto/content/pull/30307

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 1]
```diff
@@ -180,6 +192,20 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
     return False
 
 
+def is_tim_reviewer_needed(packs_in_pr: set[str]) -> bool:
+    for pack in packs_in_pr:
+        pack_metadata = get_pack_metadata(pack)
+        for pr_file in pack:
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:198]: From your code here it looks like the `pack` contains a list of files but according to `packs_to_check_in_pr` which outputs the list of `packs_in_pr` it only contains a list of strings.

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 2]
```diff
@@ -180,6 +192,20 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
...
+            if "yml" in pr_file:
+                if "feed: true" in pr_file:
+                    return True
+        tags = pack_metadata.get("tags")
+        categories = pack_metadata.get("categories")
+        if "Threat Intelligence Management" in tags or "Data Enrichment & Threat Intelligence" in categories:
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:204]: Change "Threat Intelligence Management" tag and "Data Enrichment & Threat Intelligence" category to be constants

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 3]
```diff
@@ -180,6 +192,20 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
     return False
 
 
+def is_tim_reviewer_needed(packs_in_pr: set[str]) -> bool:
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:195]:
```suggestion
def is_tim_content(packs_in_pr: set[str]) -> bool:
```

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 4]
```diff
@@ -265,6 +291,13 @@ def main():
         pr.add_to_assignees(security_reviewer)
         pr.add_to_labels(SECURITY_LABEL)
 
+    # adding TIM reviewer
+    packs_in_pr = packs_to_check_in_pr(pr_files)
+    if is_tim_reviewer_needed(packs_in_pr):
+        if support_label in (XSOAR_SUPPORT_LEVEL_LABEL, PARTNER_SUPPORT_LEVEL_LABEL):
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:297]: Create a new function called `is_tim_reviewer_needed` that will call `is_tim_content ` and will also check the conditions of the labels and categories and will return true or false

File: Utils/github_workflow_scripts/handle_external_pr.py (hunk 1 above)
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py:198]: Yes, I did several changes to this function and forgot the pr_files argument - fixed

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 5]
```diff
@@ -180,6 +197,42 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
...
+    :param packs_in_pr: packs that are part of the PR
+    :return: returns True or False if tim reviewer needed
+    """
+    for pack in packs_in_pr:
+        pack_object = BaseContent.from_path(CONTENT_PATH / pack)
+        try:
+            if isinstance(pack_object, Pack):
+                for integration in pack_object.content_items.integration:
+                    if integration.is_feed:
+                        return True
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:211]: This logic is incorrect. Imagine a PR where a non-TIM script was changed, this code will extract the pack and will check if there's an integration in this pack of TIM type and will return true. This is obviously wrong cause we don't want to involve TIM owner in such a case, we will only if the file changed belongs to an integration of type feed.

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 6]
```diff
@@ -180,6 +197,42 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
...
+        except Exception as error:
+            print(f'Received error: {error}\nwhen trying to find pack {pack}')
+        pack_metadata = get_pack_metadata(pack)
+        tags = pack_metadata.get("tags")
+        categories = pack_metadata.get("categories")
+        print("tim in categories")
```
[the remark this answers is not in the document]
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py:217]: This was part of debugging - it's not relevant at all - I just wanted to make sure where I catch the feed.
I deleted all the other prints but forgot to delete this print

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 7]
```diff
@@ -200,59 +200,31 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
...
-    #                      return True
-    # pack_metadata = get_pack_metadata(pack)
-    # tags = pack_metadata.get("tags")
-    # categories = pack_metadata.get("categories")
-    # if TIM_TAGS in tags or TIM_CATEGORIES in categories:
-    #     return True
-    # return False
+        if integration.is_feed:
+            return True
+        integrations_checked.append(integration.path)
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:221]: This should be before the if on line 219

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 8]
```diff
@@ -180,14 +200,60 @@ def is_requires_security_reviewer(pr_files: list[str]) -> bool:
...
+    according to the login in is_tim_reviewer_needed
+
+    Arguments:
+    - pr_files: List[str] The list of files changed in the Pull Request.
+
+    Returns: returns True or False if tim reviewer needed
+    """
+    integrations_checked = []
+    for file in pr_files:
+        integration = BaseContent.from_path(CONTENT_PATH / file)
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:215]: This returns the same value for both panos.py and panos.yml?

File: Utils/github_workflow_scripts/handle_external_pr.py (hunk 7 above)
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py:221]: It doesn't really matter - if the integraition is feed, the function will return true and exit, and if it's not true, i'll continue the iteration.
but I moved it before the IF

File: Utils/github_workflow_scripts/handle_external_pr.py (hunk 8 above)
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py:215]: yes - the integration name will be returned

File: Utils/github_workflow_scripts/handle_external_pr.py (hunk 7 above)
Comment (ShahafBenYakir) [reply Utils/github_workflow_scripts/handle_external_pr.py]: >but I moved it before the IF

Are you sure? it is still there

**Review by ShahafBenYakir — APPROVED** [review]

File: Utils/github_workflow_scripts/handle_external_pr.py (hunk 7 above)
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py]: image

**Pull request merged**